crates/oxc_linter/src/rules/typescript/no_misused_new.rs

```diff
@@ -0,0 +1,138 @@
+use oxc_ast::{ast::{TSSignature, TSType, TSTypeName, PropertyKey, ClassElement}, AstKind};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("typescript-eslint(no-misused-new): Interfaces cannot be constructed, only classes.")]
+#[diagnostic(
+    severity(warning)
+)]
+struct NoMisusedNewInterfaceDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("typescript-eslint(no-misused-new): Class cannot have method named `new`.")]
+#[diagnostic(
+    severity(warning)
```

Comment:
nitpick: do we have any help message for why `new` and `constructor` is bad?

Reply:
There are no corresponding help messages in the original typescript lint rules, so I'm not sure if this help message is appropriate.